File: src/com/owlrobotics/pedropathingvisualizer/MainPanel.java
package com.owlrobotics.pedropathingvisualizer;

import com.owlrobotics.pedropathingvisualizer.pedropathing.entities.BotEntity;
import com.owlrobotics.pedropathingvisualizer.pedropathing.util.XYLayout;

import javax.swing.*;
import java.awt.*;
import java.awt.image.BufferedImage;
import java.util.ArrayList;

public class MainPanel extends JPanel {

    // Variables required for com.owlrobotics.pedropathingvisualizer.ControlPanel and com.owlrobotics.pedropathingvisualizer.FieldPanel
    int fieldSize;
    BufferedImage fieldImage;
    JFrame frame;
    ArrayList<BotEntity> entities;
    Dimension planeOrigin;
    double pixelsPerInch;


    FieldPanel fieldPanel;
    ControlPanel controlPanel;

    // com.owlrobotics.pedropathingvisualizer.MainPanel class constructor
    public MainPanel(BufferedImage fieldImg, double fieldRotation, JFrame frame, ArrayList<BotEntity> entities, Dimension planeOrigin, double pixelsPerInch, int fieldSize, int targetFPS) {
        // Set the variables
        this.fieldImage = fieldImg;
        this.frame = frame;
        this.entities = entities;
        this.planeOrigin = planeOrigin;
        this.pixelsPerInch = pixelsPerInch;
        this.fieldSize = fieldSize;

        // Panel settings
        this.setBackground(Color.BLACK);
        this.setPreferredSize(new Dimension(fieldSize + 60 + 800, fieldSize + 40));
        this.setLayout(new XYLayout());
        this.setVisible(true);

        // Create fieldPanel
        fieldPanel = new FieldPanel(
                fieldImg,
                frame,
                entities,
                pixelsPerInch,
                planeOrigin,
                fieldRotation,
                fieldSize,
                targetFPS);

        // Add com.owlrobotics.pedropathingvisualizer.FieldPanel
        this.add(fieldPanel);

        // Create and add com.owlrobotics.pedropathingvisualizer.ControlPanel
        this.add(new ControlPanel(
                fieldSize,
                entities,
                fieldPanel));

    }
}

File: src/com/owlrobotics/pedropathingvisualizer/MainPanel.kt
package com.owlrobotics.pedropathingvisualizer

import com.owlrobotics.pedropathingvisualizer.pedropathing.entities.BotEntity
import com.owlrobotics.pedropathingvisualizer.pedropathing.util.XYLayout
import java.awt.Color
import java.awt.Dimension
import java.awt.image.BufferedImage
import javax.swing.JFrame
import javax.swing.JPanel

class MainPanel(
    fieldImg: BufferedImage, fieldRotation: Double,
    var frame: JFrame,
    var entities: ArrayList<BotEntity>,
    var planeOrigin: Dimension,
    var pixelsPerInch: Double, // Variables required for com.owlrobotics.pedropathingvisualizer.ControlPanel and com.owlrobotics.pedropathingvisualizer.FieldPanel
    var fieldSize: Int, targetFPS: Int
) :
    JPanel() {
    // Set the variables
    var fieldImage: BufferedImage = fieldImg


    var fieldPanel: FieldPanel
    var controlPanel: ControlPanel? = null

    // com.owlrobotics.pedropathingvisualizer.MainPanel class constructor
    init {
        // Panel settings
        this.background = Color.BLACK
        this.preferredSize = Dimension(fieldSize + 60 + 800, fieldSize + 40)
        this.layout = XYLayout()
        this.isVisible = true

        // Create fieldPanel
        fieldPanel = FieldPanel(
            fieldImg,
            frame,
            entities,
            pixelsPerInch,
            planeOrigin,
            fieldRotation,
            fieldSize,
            targetFPS
        )

        // Add com.owlrobotics.pedropathingvisualizer.FieldPanel
        this.add(fieldPanel)

        // Create and add com.owlrobotics.pedropathingvisualizer.ControlPanel
        this.add(
            ControlPanel(
                fieldSize,
                entities,
                fieldPanel
            )
        )
    }
}
